### PR TITLE
PRT-760-provider-addon-support

### PR DESCRIPTION
**Pull request by omerlavanet**

### CODE REVIEW

File: x/pairing/keeper/filters/addon_filter.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	spectypes "github.com/lavanet/lava/x/spec/types"
+)
+
+type AddonFilter struct {
+	collections map[spectypes.CollectionData]struct{}
+}
+
+func (f *AddonFilter) InitFilter(strictestPolicy planstypes.Policy) bool {
+	if len(strictestPolicy.ChainPolicies) > 0 {
+		chainPolicy := strictestPolicy.ChainPolicies[0] // only the specific spec chain policy is there
```
Comment (Yaroms) [inline x/pairing/keeper/filters/addon_filter.go:17]: why at index 0?

File: x/pairing/keeper/filters/addon_filter.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	}, utils.Attribute{
+		Key:   "endpoints",
+		Value: endpoints,
+	})
+requirementsLoop:
+	for requirements := range requirements {
+		for _, endpoint := range endpoints {
+			if endpoint.Addon == requirements.AddOn && endpoint.ApiInterface == requirements.ApiInterface {
+				// this endpoint fulfills the requirement
+				continue requirementsLoop
```
Comment (Yaroms) [inline x/pairing/keeper/filters/addon_filter.go:58]: can just use break

File: x/plans/types/policy.go [hunk 3]
```diff
@@ -8,21 +8,44 @@ import (
...
+	spectypes "github.com/lavanet/lava/x/spec/types"
 	"github.com/mitchellh/mapstructure"
 )
 
-func (policy *Policy) ContainsChainID(chainID string) bool {
-	if len(policy.ChainPolicies) == 0 {
-		// empty chainPolicies -> support all chains
-		return true
-	}
+const WILDCARD_CHAIN_POLICY = "*"
```
[the remark this answers is not in the document]
Comment (Yaroms) [reply x/plans/types/policy.go:16]: and add some explanation for what it does?

File: x/spec/keeper/spec.go [hunk 4]
```diff
@@ -264,6 +265,10 @@ func (k Keeper) GetExpectedInterfacesForSpec(ctx sdk.Context, chainID string, ma
 			)
 		}
 		expectedInterfaces = k.getExpectedInterfacesForSpecInner(&spec, expectedInterfaces, mandatory)
+	} else {
```
Comment (Yaroms) [inline x/spec/keeper/spec.go:268]: we dont need the else here, just return with the error

File: x/plans/keeper/plan.go [hunk 5]
```diff
@@ -75,3 +82,29 @@ func (k Keeper) ExportPlans(ctx sdk.Context) []commonTypes.RawMessage {
...
+		if specID == types.WILDCARD_CHAIN_POLICY && len(chainPolicy.Apis) == 0 && len(chainPolicy.Collections) == 0 {
+			continue // this is allowed
+		}
+		expectedInterfaces, err := k.specKeeper.GetExpectedInterfacesForSpec(ctx, specID, false)
+		if err != nil {
+			return err
+		}
+		for _, collection := range chainPolicy.Collections {
+			addon := collection.AddOn
+			if addon == collection.ApiInterface {
```
[the remark this answers is not in the document]
Comment (omerlavanet) [reply x/plans/keeper/plan.go:98]: optional apiInterface

**Review by ranlavanet — APPROVED** [review]

**Pull request merged**